morpheus/pipeline/pipeline.py

```diff
@@ -452,14 +451,13 @@ async def join(self):
         Suspend execution all currently running stages and the MRC pipeline.
         Typically called after `stop`.
         """
-        await self._completion_event.wait()
+        await self._post_start_future
```

Comment:
We need error handling here to ensure the future is set. This should be caught by a test